lib/client.js

```diff
@@ -78,6 +81,11 @@ module.exports = function (dependencies) {
               this.config.token.regenerate(tokenGeneration);
               resolve(this.write(notification, device, retryCount + 1));
               return;
+            } else if (status === "500" && response.reason === "InternalServerError") {
+              stream.connection.close();
+              let error = new VError("Error 500, stream ended unexpectedly");
```

Comment:
4 including `{ device }` on a `200` status.

Tricky to pin down what users would need access to, but seeing as, currently, success omits `status` or `response`, it does make sense to only include anything extraneous to `device` if it's relevant.

It may make sense to _always_ include `device` and `status` and, if `status !== 200`, include either `response` OR `error` too.